packages/material-ui/src/Modal/ModalManager.ts

```diff
@@ -84,54 +99,55 @@ function handleContainer(containerInfo, props) {
     const parent = container.parentElement;
     const containerWindow = ownerWindow(container);
     const scrollContainer =
-      parent.nodeName === 'HTML' &&
-      containerWindow.getComputedStyle(parent)['overflow-y'] === 'scroll'
+      parent?.nodeName === 'HTML' && containerWindow.getComputedStyle(parent).overflowY === 'scroll'
         ? parent
         : container;
 
     // Block the scroll even if no scrollbar is visible to account for mobile keyboard
     // screensize shrink.
     restoreStyle.push({
       value: scrollContainer.style.overflow,
-      key: 'overflow',
+      property: 'overflow',
       el: scrollContainer,
     });
     scrollContainer.style.overflow = 'hidden';
   }
 
   const restore = () => {
-    if (fixedNodes) {
-      [].forEach.call(fixedNodes, (node, i) => {
-        if (restorePaddings[i]) {
-          node.style.paddingRight = restorePaddings[i];
-        } else {
-          node.style.removeProperty('padding-right');
-        }
-      });
-    }
-
-    restoreStyle.forEach(({ value, el, key }) => {
+    restoreStyle.forEach(({ value, el, property }) => {
       if (value) {
-        el.style.setProperty(key, value);
+        el.style.setProperty(property, value);
       } else {
-        el.style.removeProperty(key);
+        el.style.removeProperty(property);
       }
     });
   };
 
   return restore;
 }
 
-function getHiddenSiblings(container) {
-  const hiddenSiblings = [];
-  [].forEach.call(container.children, (node) => {
-    if (node.getAttribute && node.getAttribute('aria-hidden') === 'true') {
-      hiddenSiblings.push(node);
+function getHiddenSiblings(container: Element) {
+  const hiddenSiblings: Element[] = [];
+  [].forEach.call(container.children, (element: Element) => {
+    if (element.getAttribute('aria-hidden') === 'true') {
+      hiddenSiblings.push(element);
     }
   });
   return hiddenSiblings;
 }
 
+interface Modal {
+  mount: Element;
+  modalRef: Element;
+}
+
+interface Container {
+  container: HTMLElement;
```

Comment:
Please mark it HTMLElement | ShadowRoot

Reply:
ShadowRoot isn't supported, as far as I know, nor we should to?

Reply:
We'd need to take a look at the Modal which passes it. I think it technically allows `Element` (`container?: Element | (() => Element | null) | null;`) which is too wide though technically fine since all the known interfaces that extend `Element` are `HTMLElement` and `SVGElement` which are fine.

The type-story is a bit unergonomic so we might go with a hack `Element` which we'll later cast to `HTMLElement | SVGElement`.